repose-aggregator/commons/commons-utilities/src/main/scala/org/openrepose/commons/utils/servlet/http/HttpServletResponseWrapper.scala

```diff
@@ -222,7 +225,25 @@ class HttpServletResponseWrapper(originalResponse: HttpServletResponse, headerMo
   override def addDateHeader(name: String, timeSinceEpoch: Long): Unit =
     addHeader(name, DateUtils.formatDate(new Date(timeSinceEpoch)))
 
+  private def logHeaderWarning(headerNames: String, mayWill: String, notStill: String, name: String, value: String): Unit = {
+    logHeaderWarning(headerNames, mayWill, notStill, s"$name: $value")
+  }
+
+  private def logHeaderWarning(headerNames: String, mayWill: String, notStill: String, nameValue: String): Unit = {
+    headerWarningLogger.info(s"Calls to $headerNames after the response has been committed $mayWill be ignored.")
+    headerWarningLogger.warn(s"The following header $mayWill $notStill arrive at the client: $nameValue")
+  }
+
+  private def ifCommitted(name: String, value: String): Unit = {
+    if (isCommitted) {
+      logHeaderWarning(
+        "addHeader, addIntHeader, addDateHeader, setHeader, setIntHeader, setDateHeader, appendHeader, or replaceHeader",
+        May, Not, name, value)
+    }
+  }
```

Comment:
You created a few methods here for the sake of logging. I believe that was done to keep the code DRY, but personally, I don't believe that is worth the additional cognitive overhead and method calls on the stack. Additionally, this is a fairly rigid structure -- the log message must always be mostly the same when these methods are used.

If you don't want to repeat the log message everywhere, which is reasonable, I would propose, instead, that you use a string template for the log message in question. Doing so would let you keep your methods functional, and provide flexibility if we ever wanted to log something else when the response has already been committed. Something like:
`private def headerWarningMessage(method: String, headerName: String): String = s"Calls to $method after the response has been committed may be ignored -- the following header may not be modified: $headerName"`

I also wouldn't bother with the `ifCommitted` method since it only buys you one line per call site, and forces you to be vague in what method was called. Also, in `removeHeader` you still end up logging the header in a different way, which adds a little inconsistency (again, because of the rigidity).

In fact, I may also condense the log messages themselves into one message. Aren't the `info` and `warn` message saying nearly the same thing? If the header call is ignored, doesn't it directly follow that the header (the reason for the call) may/will not make it to the client?

So with all of that, I would probably end up with something like:
```
...

override def addHeader(name: String, value: String): Unit = {
  if (isCommitted) {
    headerWarningLogger.warn(headerWarningMessage("addHeader", name) + ": " + value)
  }

  ...
}

override def removeHeader(name: String, value: String): Unit = {
  ifMutable(headerMode) {
    if (isCommitted) {
      headerWarningLogger.warn(headerWarningMessage("removeHeader", name))
    }

    ...
  }
}

...
```

You might want to change a few words or parameters from what I have if you want to maintain the clarity that is currently being provided, but that should be easy enough to do using this structure.